scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,980 @@
+    local charName = player:getName()
+
+    if
+        npcUtil.tradeHasExactly(trade, {
+            xi.events.egg_hunt.charToEgg(string.sub(charName, 1, 1)),
+            xi.events.egg_hunt.charToEgg(string.sub(charName, 2, 2)),
+            xi.events.egg_hunt.charToEgg(string.sub(charName, 3, 3)),
+        })
+    then
+        -- If not 2007, or player already has Egg Helm,.minor reward is issued instead
+        if settings.ERA_2007 and not player:hasItem(xi.items.EGG_HELM) then
+            player:setVar(settings.VAR.FIRST_THREE, 1)
+            return xi.items.EGG_HELM
+        end
+
+        player:setVar(settings.VAR.FIRST_THREE, 1)
+        return minorReward()
+    end
+end
+
+local sevenKind = function(player, npc, trade)
+    -- First three required
+    if not hasFirstThree(player) then
+        return
+    end
+
+    for letterOffset = 0, 26 do
+        if npcUtil.tradeHasExactly(trade, { { xi.items.A_EGG + letterOffset, 7 } }) then
+            if not player:hasItem(xi.items.FORTUNE_EGG) then
+                return xi.items.FORTUNE_EGG
+            end
+
+            if settings.MINOR_REWARDS then
+                return minorReward()
+            end
+
+            return
+        end
+    end
+end
+
+local straightEight = function(player, npc, trade)
+    -- First three required
+    if not hasFirstThree(player) then
+        return
+    end
+
+    local initial = string.lower(string.sub(player:getName(), 1, 1)) -- eg. "a"
+    local letter = string.byte(initial) - 97 + xi.items.A_EGG        -- itemID
+    local eggs = {}
+
+    for i = 1, 8 do
+        table.insert(eggs, letter)
+        letter = letter + 1
+
+        if letter > xi.items.Z_EGG then
+            letter = xi.items.A_EGG
+        end
+    end
+
+    if npcUtil.tradeHasExactly(trade, eggs) then
+        if not player:hasItem(xi.items.HAPPY_EGG) then
+            return xi.items.HAPPY_EGG
+        end
+
+        if settings.MINOR_REWARDS then
+            return minorReward()
+        end
+    end
+end
+
+local regionNames =
+{
+    xi.events.egg_hunt.stringToEggs("RONFA"),
+    xi.events.egg_hunt.stringToEggs("ZULKH"),
+    xi.events.egg_hunt.stringToEggs("NORVA"),
+    xi.events.egg_hunt.stringToEggs("GUSTA"),
+    xi.events.egg_hunt.stringToEggs("DERFL"),
+    xi.events.egg_hunt.stringToEggs("SARUT"),
+    xi.events.egg_hunt.stringToEggs("KOLSH"),
+    xi.events.egg_hunt.stringToEggs("ARAGO"),
+    xi.events.egg_hunt.stringToEggs("FAURE"),
+    xi.events.egg_hunt.stringToEggs("VALDE"),
+    xi.events.egg_hunt.stringToEggs("QUFIM"),
+    xi.events.egg_hunt.stringToEggs("LITEL"),
+    xi.events.egg_hunt.stringToEggs("KUZOT"),
+    xi.events.egg_hunt.stringToEggs("VOLLB"),
+    xi.events.egg_hunt.stringToEggs("ELSHI"),
+    xi.events.egg_hunt.stringToEggs("ELSHI"),
+    xi.events.egg_hunt.stringToEggs("TULIA"),
+    xi.events.egg_hunt.stringToEggs("MOVAL"),
+    xi.events.egg_hunt.stringToEggs("TAVNA"),
+}
+
+local beastCostumes =
+{
+    580, -- Yagudo
```

Comment:
TODO: Someone else for another time, should we canonicalize costumes/models into their own enum?

Reply:
> TODO: Someone else for another time, should we canonicalize costumes/models into their own enum?

it would be far too massive, with a lot of difficulty in establishing correct names. if that could happen I would say it'd be at minimum 5+ years down the road as I already embarked down mapping them out on the wiki ages ago.